tests/src/nodes/nomos.rs

```diff
@@ -263,13 +263,12 @@ fn create_node_config(
         },
         consensus: CarnotSettings {
             private_key,
-            overlay_settings: FlatOverlaySettings {
+            overlay_settings: TreeOverlaySettings {
                 nodes,
                 leader: RoundRobin::new(),
-                // By setting the leader_threshold to 1 we ensure that all nodes come
-                // online before progressing. This is only necessary until we add a way
-                // to recover poast blocks from other nodes.
-                leader_super_majority_threshold: Some(threshold),
+                current_leader: [0; 32].into(),
+                number_of_committees: 1,
+                committee_membership: RandomBeaconState::initial_sad_from_entropy([0; 32]),
```

Comment:
I've tested the `ten_nodes_happy` with disabling mixnet and revert the `unreachable!()`, in order to check the reason why local_timeout occurs in the happy path test. (I reverted the `unreachable!()` because the stream is closed when local_timeout occurs in our case).

**In short, local_timeout occurs due to the design of our test environment, only when we use tree-overlay with random-beacon.**

For example, there are 10 nodes (node-0~9). Let’s say the next leader (of view 1) is node-6, selected by random-beacon. In our test environment, 10 nodes are started sequentially (from 0 to 9). In other words, node-6 is not started until node-5 becomes ready.

In this case, I saw that node-6 received only 4 votes for view 0, and all of those votes were from node-6\~9 which were started after node-6. **That is, votes from node-0~5 (started before node-6) weren’t delivered to the node-6.**

In fact, this is the issue that we’re already aware of. Our current libp2p network backend doesn’t provide a way to catch-up past messages. 

Btw, this panic didn’t happen in the previous test with flat-overlay because the round-robin is used instead of random-beacon. That is, the next leader (of view 1) is node-0 which is started at first. So, this is no problem for node-0 to receive votes from all other nodes that are started after itself.

To make the test work with tree-overlay + random-beacon, we may need to pause the consensus service of all nodes until all 10 nodes are started successfully. Actually, we have a 1s `sleep` before starting the consensus service: https://github.com/logos-co/nomos-node/pull/338, but this might not be enough. I'm thinking about other solutions.

Reply:
Nice job!! all this seems about right. Now we need to tackle it so everything is clear next time.

Reply:
Next steps discussed:
- Modify the test to check if all blocks are happy block
- Modify the test to start the node first who is going to be the next leader (instead of always starting the node-0 first)

Reply:
Ah.. I just found that we're using both `RoundRobin` and `RandomBeaconState`. So, let me rephrase the sentence above to "We use `RoundRobin` for `LeaderSelection` but the node list is suffled at the beginning (and whenever overlay is refreshed) by `CommitteeMembership` which is `RandomBeacon`."

Reply:
Resolved by https://github.com/logos-co/nomos-node/pull/425